delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala

```diff
@@ -264,8 +264,11 @@ object Resources {
     * Expands the segment to a [[ResourceRef]]
     */
   def expandResourceRef(segment: IdSegment, context: ProjectContext): IO[ResourceRef] =
+    expandResourceRef(segment, context, InvalidResourceId)
+
+  def expandResourceRef(segment: IdSegment, context: ProjectContext, notFound: String => Throwable): IO[ResourceRef] =
     IO.fromOption(segment.toIri(context.apiMappings, context.base).map(ResourceRef(_)))(
-      InvalidResourceId(segment.asString)
+      notFound(segment.asString)
     )
```

Comment:
Ok for me but why the second method here?

Reply:
Ah yeah, was going to reuse it in the ES module but forgot to call it - will open another PR as I think it's a bit bigger